File: trees/dfs.go
/*
 * Given a tree with n children per node, capture the dfs traversal in an array.
 */
package main

type Node struct {
	Name     string
	Children []*Node
}

//O(v+e) time and O(v) space - v-vertices and e-edges
func (n *Node) DepthFirstSearch(array []string) []string{
	array=append(array,n.Name)
	for _,child := range n.Children {
			array = child.DepthFirstSearch(array)
	}
	return array
}

File: trees/dfs.py
/*
 * Given a tree with n children per node, capture the dfs traversal in an array.
 */
 class Node:
    def __init__(self, name):
        self.children = []
        self.name = name

    def addChild(self, name):
        self.children.append(Node(name))
        return self

	//time O(v+e) and space O(v) - v-vertices and e -edges
	def depthFirstSearch(self, array):
	    array.append(self.name)
		for a in self.children:
			a.depthFirstSearch(array)
		return array
